### PR TITLE
Lint progress

### PR DESCRIPTION
**Pull request by nathanielmanistaatgoogle**
Can you spot the bugfix?

### CODE REVIEW

File: src/python/grpcio/grpc/beta/implementations.py [hunk 1]
```diff
@@ -217,7 +217,7 @@ def dynamic_stub(channel, service, cardinalities, options=None):
   Returns:
     A face.DynamicStub with which RPCs can be invoked.
   """
-    effective_options = StubOptions() if options is None else options
+    effective_options = _EMPTY_STUB_OPTIONS if options is None else options
```
Comment (kpayson64) [inline src/python/grpcio/grpc/beta/implementations.py]: Its almost like we have a compiler now.

Comment (nathanielmanistaatgoogle) [inline src/python/grpcio/grpc/beta/implementations.py]: I know, and just a few weeks ago @mehrdada was wondering what all the fuss was about!

Comment (mehrdada) [inline src/python/grpcio/grpc/beta/implementations.py]: Oh well... True believer in static analysis I have always been. Was just too charitable to Python users incorrectly assuming they _must_ have some magic powers, hence their choice of the language. Burned by optimism again! _C'est la vie_... 🏳

**Comment by nathanielmanistaatgoogle** [comment]
Pre-merge test assessment:

* Basic Tests Linux: [9903](https://github.com/grpc/grpc/issues/9903).
* Basic Tests Mac: [9804](https://github.com/grpc/grpc/issues/9804).
* Bazel Full Build & Tests: not yet in effect.
* gRPC_interop_pull_requests: [9937](https://github.com/grpc/grpc/issues/9937).
* kokoro: not yet in effect.

**Pull request merged**